homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,427 @@
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        reducer=Reducers.mean,
+    ),
+    AmbientNetworkSensorEntityDescription(
+        key=TYPE_YEARLYRAININ,
+        translation_key="yearly_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        suggested_display_precision=2,
+        reducer=Reducers.mean,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Ambient Network sensor platform."""
+
+    coordinator: AmbientNetworkDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is not None:
+        entities: list[AmbientNetworkSensor] = []
+        for description in SENSOR_DESCRIPTIONS:
+            # Check whether any of the stations report this sensor value.
+            for station_data in coordinator.data.values():
+                sensor_keys: list[str]
+                if description.key == TYPE_FEELSLIKE:
+                    # Feels like is calculated from temperature, relative humidity, and
+                    # wind speed.
+                    sensor_keys = [TYPE_TEMPF, TYPE_HUMIDITY, TYPE_WINDSPEEDMPH]
+
+                elif description.key == TYPE_DEWPOINT:
+                    # Dew point is calculated from temperature and relative humidity.
+                    sensor_keys = [TYPE_TEMPF, TYPE_HUMIDITY]
+
+                else:
+                    # Everything else is calculated from the reported value directly.
+                    sensor_keys = [description.key]
+                if all(
+                    AmbientNetworkSensor.get_sensor_value(station_data, key) is not None
+                    for key in sensor_keys
+                ):
+                    entities.append(
+                        AmbientNetworkSensor(
+                            coordinator,
+                            description,
+                            coordinator.config_entry.data[ENTITY_MNEMONIC],
+                        )
+                    )
+                    break
+        async_add_entities(entities)
+
+
+class AmbientNetworkSensor(AmbientNetworkEntity, SensorEntity):
+    """A sensor implementation for an Ambient Weather Network sensor."""
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: AmbientNetworkSensorEntityDescription,
+        mnemonic: str,
+    ) -> None:
+        """Initialize a sensor object."""
+
+        super().__init__(coordinator, description, mnemonic)
+        # Override the entity_id to make them cleaner (otherwise Homeassistant
+        # will name them _precipitation_1, _precipitation_2, etc.)
+        self.entity_id = f"sensor.{self._device_id.lower()}_{description.key}"
```

Comment:
No? Have you generated translations?

Reply:
Yes. Translations are set up [here](https://github.com/home-assistant/core/pull/105779/files#diff-826a69d62aa78ee1eba2a0563411e60ab894202e56847985f6658b537f3b3e8cR40).

Reply:
Removed and fixed.